File: LC977.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        
        vector <int> sq(nums);

        int l = 0, r = nums.size() - 1;

        for(int k = nums.size()-1; k >= 0; k--)
            if(abs(nums[l]) > abs(nums[r]))
                sq[k] = nums[l]*nums[l++];
            else
                sq[k] = nums[r]*nums[r--];        

        return sq;
    }

};

File: LC977.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:

        l = [0]*len(nums)

        left = 0
        r = len(nums)-1

        for k in range(len(nums)-1, -1, -1):
            if abs(nums[left]) > abs(nums[r]):
                l[k] = nums[left]**2
                left += 1
            else:
                l[k] = nums[r]**2
                r -= 1
            
        return l
